README.md

```diff
@@ -123,6 +123,118 @@ if mti != "0810" {
 }
 ```
 
+## Connection `Pool`
+
+Sometimes you want to establish connections to multiple servers and re-create
+them when such connections are closed due to a network errors. Connection `Pool`
+is really helpful for such use cases.
+
+To use `Pool` first, you need to create factory function that knows how to create
+connections and list of adresses you want to establish connection with.
+
+```go
+// Factory method that will build connection
+factory := func(addr string) (*connection.Connection, error) {
+	c, err := connection.New(
+		addr,
+		testSpec,
+		readMessageLength,
+		writeMessageLength,
+		// set shot connect timeout so we can test re-connects
+		connection.ConnectTimeout(500*time.Millisecond),
+		connection.OnConnect(func(c *connection.Connection) {
```

Comment:
Any reason this cannot be hooked into these connections by default instead of calling it out here? Seems like unnecessary boilerplate

Reply:
Do you mean setting status or what? In real clients, we will send network management messages in `OnConnect`, and based on the result, we will set connection status and return nil/error.

Reply:
Makes sense, resolve please :)